src/analysis/retail/priest/shadow/modules/talents/Voidweaver/Voidheart.tsx

```diff
@@ -0,0 +1,48 @@
+import TALENTS from 'common/TALENTS/priest';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { DamageEvent } from 'parser/core/Events';
+import Events from 'parser/core/Events';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+import { VOIDHEART_MULTIPLIER } from '../../../constants';
+
+class Voidheart extends Analyzer {
+  damage = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.VOIDHEART_TALENT);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+  }
+
+  onDamage(event: DamageEvent) {
+    if (this.selectedCombatant.hasBuff(SPELLS.SHADOW_PRIEST_VOIDWEAVER_ENTROPIC_RIFT_BUFF)) {
+      //console.log(event.ability.name,"Spell of type", event.ability.type)
+      //During Entropic Rift, if the ability that caused the damage event is a shadow spell, we have a damage increase.
+      //shadow damage is type 32.
+      //the multi school spells with shadow should also be increased, but the only one this spec has access to is shadowfrost (48)
+      if (event.ability.type === 32 || event.ability.type === 48) {
```

Comment:
you can use the `MAGIC_SCHOOL` helpers to handle this:

```suggestion
      //During Entropic Rift, if the ability that caused the damage event is a shadow spell, we have a damage increase. This also applies to multi-school spells (like Shadowfrost spells)
      if (isMatchingDamageType(event.ability.type, MAGIC_SCHOOLS.ids.SHADOW)) {
```

important note: if there are spells in Shadow's kit that *change types*, those changes are *not* reflected on WCL for technical reasons. for the purposes of Voidheart that is probably fine, but Ret has had some issues before.

Reply:
That is a good change, and thankfully shadow doesn't have anything like that.  I'll leave a comment in there as a reminder just in case that changes in the future.